zeppelin-zengine/src/main/java/org/apache/zeppelin/notebook/repo/torrent/BittorrentNotebookRepo.java

```diff
@@ -0,0 +1,257 @@
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package org.apache.zeppelin.notebook.repo.torrent;
+
+import com.frostwire.jlibtorrent.TorrentHandle;
+import com.frostwire.jlibtorrent.alerts.StateChangedAlert;
+
+import org.apache.commons.vfs2.FileObject;
+import org.apache.commons.vfs2.FileType;
+import org.apache.commons.vfs2.NameScope;
+import org.apache.zeppelin.conf.ZeppelinConfiguration;
+import org.apache.zeppelin.notebook.repo.VFSNotebookRepo;
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+
+import java.io.File;
+import java.io.FileNotFoundException;
+import java.io.FileOutputStream;
+import java.io.IOException;
+import java.io.InputStream;
+import java.io.OutputStream;
+import java.util.List;
+import java.util.concurrent.CopyOnWriteArrayList;
+
+
+/**
+ * TO download note from magnet link and share a note
+ */
+public class BittorrentNotebookRepo extends VFSNotebookRepo implements TorrentEngineListener {
+  private static final Logger LOG = LoggerFactory.getLogger(BittorrentNotebookRepo.class);
+  static File torrentHomeDir;
+  static File torrentDir;
+  static File noteDownloadDir;
+  static File resumeDataDir;
+  static File settingsFile;
+  private List<TorrentSocket> connections = new CopyOnWriteArrayList<>();
+  private TorrentEngine engine;
+
+  public BittorrentNotebookRepo(ZeppelinConfiguration conf) throws IOException {
+    super(conf);
+    torrentHomeDir = new File(getRootDir().getName().getPath());
+    torrentDir = new File(torrentHomeDir.getAbsolutePath() + "/" + "torrents");
+    noteDownloadDir = new File(torrentHomeDir.getAbsolutePath() + "/" + "notes");
+    resumeDataDir = new File(torrentHomeDir.getAbsolutePath() + "/" + "resume");
+    settingsFile = new File(torrentHomeDir.getAbsolutePath() + "settings.dat");
+
+    loadLibrary();
+    setupDir();
+    engine = TorrentEngine.Instance;
+    engine.start();
+    engine.setListener(this);
+  }
+
+  private static void loadLibrary() throws IOException {
+    String path = LIBTORRENT_OS_LIBRARY_PATH();
+    if (path == null) {
+      LOG.error("Unknown Operating system");
+      throw new IOException();
+    }
+    InputStream is = BittorrentNotebookRepo.class.getResourceAsStream(path);
+    File temp = File.createTempFile("libjlibtorrent", ".so");
+    OutputStream os = new FileOutputStream(temp);
+    if (is == null) {
+      throw new FileNotFoundException("File " + path + " was not found inside JAR.");
+    }
+    int readBytes;
+    byte[] buffer = new byte[1024];
+    try {
+      while ((readBytes = is.read(buffer)) != -1) {
+        os.write(buffer, 0, readBytes);
+      }
+    } finally {
+      os.close();
+      is.close();
+    }
+    //System.load(temp.getAbsolutePath());
+    System.setProperty("jlibtorrent.jni.path", temp.getAbsolutePath());
+  }
+
+  private static String LIBTORRENT_OS_LIBRARY_PATH() {
```

Comment:
please make sure you follow the project java style guide regarding [method naming](https://google.github.io/styleguide/javaguide.html#s5.2.3-method-names) conventions